src/scancode/cli.py

```diff
@@ -561,7 +562,7 @@ def _scanit(paths, scanners, scans_cache_class, diag, timeout=DEFAULT_TIMEOUT):
             if scan_result.get('scan_errors'):
                 success = False
 
-    return success, rel_path
+    return success, rel_path, scan_result
```

Comment:
In returning the results here, I think that you are bypassing the whole caching mechanics.